jwst/extract_1d/extract.py

```diff
@@ -3653,32 +1545,22 @@ def create_extraction(extract_ref_dict,
             source_type = input_model.meta.target.source_type
 
     # Turn off use_source_posn if the source is not POINT
-    if source_type != 'POINT':
-        use_source_posn = False
-        log.info(f"Setting use_source_posn to False for source type {source_type}")
+    if source_type != 'POINT' or exp_type in WFSS_EXPTYPES:
+        if kwargs.get('use_source_posn') is None:
+            kwargs['use_source_posn'] = False
+            log.info(f"Setting use_source_posn to False for exposure type {exp_type}, "
+                     f"source type {source_type}")
```

Comment:
@hayescr - This is the fix for JP-3813, to allow an explicit setting of use_source_posn=True even when the source_type is not POINT.

Reply:
Looks good to me, I checked for a NIRSpec example and it no longer overrides the parameter.